cedar-policy/src/api/err.rs

```diff
@@ -115,6 +116,197 @@ pub enum ReAuthorizeError {
+}
+
+impl UndeclaredSchemaConstructError {
+    /// Getter of schema construct kind
+    pub fn get_kind(&self) -> SchemaConstructKind {
+        self.kind.clone()
+    }
+
+    /// Getter of schema constructs
+    pub fn get_constructs(&self) -> impl Iterator<Item = SmolStr> + '_ {
+        self.constructs.iter().cloned()
+    }
+}
+
+/// Undeclared schema construct error
+#[derive(Debug, Clone, Diagnostic, Error)]
+#[error("duplicate {kind} `{construct}`")]
+pub struct DuplicateSchemaConstructError {
+    kind: SchemaConstructKind,
+    construct: SmolStr,
+}
+
+impl DuplicateSchemaConstructError {
+    /// Getter of schema construct kind
+    pub fn get_kind(&self) -> SchemaConstructKind {
+        self.kind.clone()
+    }
+
+    /// Getter of schema constructs
+    pub fn get_construct(&self) -> SmolStr {
+        self.construct.clone()
+    }
+}
+
+/// Cycle in action hierarchy error
+#[derive(Debug, Clone, Diagnostic, Error)]
+#[error("cycle in action hierarchy containing `{node}`")]
+pub struct CycleInActionHierarchyError {
+    node: EntityUid,
+}
+
+impl CycleInActionHierarchyError {
+    /// Getter of the action in the cycle
+    pub fn get_node_in_cycle(&self) -> SmolStr {
+        self.node.to_smolstr()
+    }
+}
+
+/// Cycle in common type references error
+#[derive(Debug, Clone, Diagnostic, Error)]
+#[error("cycle in common type references containing `{node}`")]
+pub struct CycleInCommonTypeReferencesError {
+    node: Name,
+}
+
+impl CycleInCommonTypeReferencesError {
+    /// Getter of the common type id in the cycle
+    pub fn get_node_in_cycle(&self) -> SmolStr {
+        self.node.to_smolstr()
+    }
+}
+
+/// Context or entity shape is not record error
+#[derive(Debug, Clone, Diagnostic, Error)]
+#[error("{context_or_shape} is declared with a type other than `Record`")]
+pub struct ContextOrShapeNotRecordError {
+    context_or_shape: ContextOrShape,
+}
+
+impl ContextOrShapeNotRecordError {
+    /// Getter of the context or shape
+    pub fn get_context_or_shape(&self) -> SmolStr {
+        self.context_or_shape.to_smolstr()
+    }
+}
+
+/// Action attribute contains empty set error
+#[derive(Debug, Clone, Diagnostic, Error)]
+#[error("action `{action_uid}` has an attribute that is an empty set")]
+pub struct ActionAttributesContainEmptySetError {
+    action_uid: EntityUid,
+}
+
+impl ActionAttributesContainEmptySetError {
+    /// Getter of the action uid
+    pub fn get_action(&self) -> SmolStr {
+        self.action_uid.to_smolstr()
+    }
+}
+
+/// Unsupported action attribute error
+#[derive(Debug, Clone, Diagnostic, Error)]
+#[error(
+    "action `{action_uid}` has an attribute with unsupported JSON representation: {escape_type}"
+)]
+pub struct UnsupportedActionAttributeError {
+    action_uid: EntityUid,
+    escape_type: SmolStr,
+}
+
+impl UnsupportedActionAttributeError {
+    /// Getter of the action uid
+    pub fn get_action(&self) -> SmolStr {
+        self.action_uid.to_smolstr()
+    }
```

Comment:
Would it make sense to return the `EntityUid` here?

Reply:
Done.